In [9]:
import kaggle

# Envoi du fichier final et récupération du résultat

In [10]:
competition = 'house-prices-advanced-regression-techniques'

In [11]:
submission_file = "./csv/prediction.csv"

In [12]:
kaggle.api.competition_submit(submission_file, "10-R-pyt-with-keras", competition)

100%|██████████| 21.1k/21.1k [00:04<00:00, 5.04kB/s]


Successfully submitted to House Prices: Advanced Regression Techniques

In [22]:
kaggle.api.competitions_submissions_list(competition)

[{'ref': 13862339,
  'totalBytes': 21635,
  'date': '2019-12-19T14:41:19.857Z',
  'description': '10-R-pyt-with-keras',
  'errorDescription': None,
  'fileName': 'prediction.csv',
  'publicScore': '0.16165',
  'privateScore': None,
  'status': 'complete',
  'submittedBy': 'laurent33',
  'submittedByRef': 'laurent33',
  'teamName': 'laurent33',
  'type': 'standard',
  'url': 'https://www.kaggle.com/submissions/13862339/13862339.raw'},
 {'ref': 13641192,
  'totalBytes': 21660,
  'date': '2019-12-05T15:14:15.28Z',
  'description': '03-keras-with-quant-and-epo300',
  'errorDescription': None,
  'fileName': 'test_predict.csv',
  'publicScore': '0.15527',
  'privateScore': None,
  'status': 'complete',
  'submittedBy': 'laurent33',
  'submittedByRef': 'laurent33',
  'teamName': 'laurent33',
  'type': 'standard',
  'url': 'https://www.kaggle.com/submissions/13641192/13641192.raw'},
 {'ref': 13641098,
  'totalBytes': 21692,
  'date': '2019-12-05T15:08:42.23Z',
  'description': '03-keras-with-q

# Historiques scores house-prices (top score = 0.07)

- score de 0.177 :
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = suppression des lignes
    - modèle entraîné sur 80% Trainfull (score obtenu de 0.138 sur set de validation 20% Train)
    - traitement des données manquantes sur Test = mise à 0 si quantitative, 'None' si qualitative
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb2 : traitements différents sur données d'entraînement et données de test (l'on ne peut pas supprimer d'input sur ce dernier car il faut faire une prédiction)
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.168 => **amélioré**
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = **mise à 0 si quantitative, 'None' si qualitative**
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.152** sur set de validation 20% Train)
    
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.168 => *pas de changement*
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = mise à la **médiane** si quantitative, 'None' si qualitative
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.155** sur set de validation 20% Train)
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.182 => *moins bon*
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = mise à la médiane si quantitative, 'None' si qualitative
    - suppression de 7 features corrélées > **0.5** (au lieu 0.7 précédemment)
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.18** sur set de validation 20% Train)
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.152 => **amélioré**
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = mise à la médiane si quantitative, 'None' si qualitative
    - les 5 features corrélées > 0.7 **non supprimées**
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.14** sur set de validation 20% Train)
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>
        - Pb3 : variables considées à tort comme quantitatives (MSSubClass)

- score de 0.153 => pas d'amélioration
    - préparation des données sur 100% du Train puis split
    - traitement des données manquantes sur Train = mise à la médiane si quantitative, 'None' si qualitative
    - les 5 features corrélées > 0.7 non supprimées
    - MSSubClass convertie en variable **qualitative**
    - modèle entraîné sur 80% Trainfull (score obtenu de **0.15** sur set de validation 20% Train)
        - Pb1 : préparation fait sur 100% du trainfull au lieu de se restreindre au train <br>

- score de 0.162
    - utilisation d'un réseau de neurones avec couches cachées